typescript/src/services/redemptions/redemptions-service.ts

```diff
@@ -149,6 +164,81 @@ export class RedemptionsService {
     return { targetChainTxHash, walletPublicKey }
   }
 
+  /**
+   * Requests a redemption of TBTC v2 token into BTC using a custom integration.
+   * The function builds the redemption data and handles the redemption request
+   * through the provided redeemer proxy.
+   * @param bitcoinRedeemerAddress Bitcoin address the redeemed BTC should be
+   *        sent to. Only P2PKH, P2WPKH, P2SH, and P2WSH address types are supported.
+   * @param amount The amount to be redeemed with the precision of the tBTC
+   *        on-chain token contract.
+   * @param l2ChainName The name of the L2 chain to request redemption on.
+   * @returns Object containing:
+   *          - Target chain hash of the request redemption transaction
+   *            (for example, Ethereum transaction hash)
+   */
+  async requestCrossChainRedemption(
+    bitcoinRedeemerAddress: string,
+    amount: BigNumber,
+    l2ChainName: DestinationChainName
+  ): Promise<{ targetChainTxHash: Hex }> {
+    const crossChainContracts = this.#crossChainContracts(l2ChainName)
+    if (!crossChainContracts || !crossChainContracts.l2BitcoinRedeemer) {
+      throw new Error(
+        `Cross-chain redeemer contracts for ${l2ChainName} not initialized`
+      )
+    }
+
+    const redeemerOutputScript = await this.getRedeemerOutputScript(
+      bitcoinRedeemerAddress
+    )
+    const nonce = Math.round(Math.random() * 10000)
```

Comment:
Is there a reason to limit the `nonce` to `[1, 10 000]` range?

```suggestion
    // guarantees uniqueness over time
    const nonce = Date.now() * 1000 + Math.floor(Math.random() * 1000)
```

Reply:
No specific requirement drove the 1‑10 000 limit, I picked it as a way to get a random integer, but it doesn’t guarantee uniqueness across rapid successive calls. Your timestamp‑based approach is more robust because it practically eliminates collisions. I updated the code accordingly by accepting your suggestion.